[Monitor Query] Add samples/docs for running big queries and overcoming API limits for size and record count (#35704)
Azure/azure-sdk-for-java: sdk/monitor/azure-monitor-query/src/samples/java/README.md

## Examples

> The following sections provide code samples covering common operations with Azure Monitor Logs and Metrics query client
> libraries.

Reviewer: ```suggestion
The following sections provide code samples covering common operations with the Azure Monitor Query client library.
```

Troubleshooting steps can be found [here][SDK_README_TROUBLESHOOTING].

## Run large log queries using Log Analytics

Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
using different time ranges.

This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 

**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few million records per hour. For larger data sets, [exporting][logs_data_export] is recommended.

This sample shows how to partition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
row-based partitioning and byte-size partitioning. The partitioning is based on the timestamp "TimeGenerated". 

> This sample is suitable for simple data retrieval queries that utilize a subset of KQL operators. The subset of KQL operators supported are can
> be found [here][kql_language_subset].

Reviewer: ```suggestion
This sample is suitable for simple data retrieval queries that utilize a subset of KQL operators. The subset of supported KQL operators can be found [here][kql_language_subset].
```

Troubleshooting steps can be found [here][SDK_README_TROUBLESHOOTING].

## Run large log queries using Log Analytics

Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
using different time ranges.

This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 

> **Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few millions records per hour. For larger data sets, [exporting][logs_data_export] is recommended.

Reviewer: ```suggestion
**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few million records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
```

> This sample shows how to parition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 

Reviewer: ```suggestion
This sample shows how to partition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
```

Troubleshooting steps can be found [here][SDK_README_TROUBLESHOOTING].

> ## Running Large Log Queries using Log Analytics

Reviewer: ```suggestion
## Run large log queries using Log Analytics
```

Due to Log Analytics [service limits][monitor_service_limits], sometimes it may
not be possible to retrieve all the expected data in a single query. For example, the number of rows returned or the maximum size of the
data returned may exceed the stated limits. One approach for overcoming these limits is to split the queries into multiple smaller queries
using different time ranges.

> This work-around allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 

Reviewer: ```suggestion
This workaround allows you to avoid the cost of exporting data to a storage account (and potentially the cost of the storage account as well). 
```

> **Disclaimer:** This approach of splitting data retrieval into smaller queries is good when dealing with a few GBs of data or a few millions
> of records per hour. For larger data sets,
> [exporting][logs_data_export] is recommended.

Reviewer: ```suggestion
**Disclaimer:** This approach of splitting data retrieval into smaller queries is useful when dealing with a few GBs of data or a few millions records per hour. For larger data sets, [exporting][logs_data_export] is recommended.
```

This sample shows how to parition a large query into smaller queries using the `LogsBatchQuery` class. The sample provides examples for 
row-based partitioning and byte-size partitioning. The partitioning is based on the timestamp "TimeGenerated". 

> This sample is suitable for simple data retrieval queries that utilize a subset of KQL known as "Reduced KQL". It contains the following

Reviewer: Is there a doc that lists the operators? If so, link to that instead of listing them here.

Reviewer: In that case, it's best to not invent the term here. Let's remove "known as Reduced KQL" from the sentence.

Reviewer 2: Here's the link I used which references this subset of operators: https://learn.microsoft.com/azure/azure-monitor/logs/basic-logs-query?tabs=portal-1#kql-language-limits
